recipes/boost/bld.bat

```diff
@@ -0,0 +1,48 @@
+:: Set the right msvc version according to Python version
+REM write a temporary batch file to map cl.exe version to visual studio version
+echo @echo 15=9 > msvc_versions.bat
+echo @echo 16=10 >> msvc_versions.bat
+echo @echo 17=11 >> msvc_versions.bat
+echo @echo 18=12 >> msvc_versions.bat
+echo @echo 19=14 >> msvc_versions.bat
+
+REM Run cl.exe to find which version our compiler is
+for /f "delims=" %%A in ('cl /? 2^>^&1 ^| findstr /C:"Version"') do set "CL_TEXT=%%A"
+FOR /F "tokens=1,2 delims==" %%i IN ('msvc_versions.bat') DO echo %CL_TEXT% | findstr /C:"Version %%i" > nul && set VSTRING=%%j && goto FOUND
+EXIT 1
+:FOUND
+
+call :TRIM VSTRING %VSTRING%
```

Comment:
conda-build takes care of that.  The change I asked @jakirkham to make basically gets out of the way if the user wants to break that relationship.  Conda-build calls the vcvarsall.bat of the compiler you tell it (or the defaults line up with Python versions).  It does that before bld.bat is called.  This script uses whichever cl.exe is on PATH to figure out the corresponding VS version.

I have no idea why that for loop mess is necessary to read output from programs, but that is all I've been able to find.  I feel dirty every time I do that (and I also have to spend a fair bit of time on Stack Overflow every time I add a new variant).